Adds benchmarks testing

chentex opened the pull request:
Signed-off-by: Vicente Zepeda Mas <chente.z.m@gmail.com>

Adds benchmark test to methods.

fixes #21

chentex committed:
Adds benchmarks testing. fixes #21

Signed-off-by: Vicente Zepeda Mas <chente.z.m@gmail.com>

chentex committed:
Adds benchmarks testing. fixes #21

Signed-off-by: Vicente Zepeda Mas <chente.z.m@gmail.com>

rlespinasse reviewed (changes requested):
The benchmark need to run on the same value to be effective, `rand.Intn` can't be use during the benchmark loop.

See https://dave.cheney.net/2013/06/30/how-to-write-benchmarks-in-go

chentex commented:
Thank you for that information @rlespinasse, I've been doing Benchmarks wrong! I'll fix it and update the PR.

rlespinasse committed:
Better tests functions (#25)

* [#20] Add switch-case tests for Byte Array operations

- ExtractBytes learn to manage edge-case of positions
- compute and trim become private functions
- Change variables name in test functions to be compliant
- Print bytes correctly during testing

Closes #20

rlespinasse committed:
[#19] Add MIT License

Closes #19

chentex committed:
Corrects benchmark test to iterate over the same value.

Signed-off-by: Vicente Zepeda Mas <chente.z.m@gmail.com>

chentex committed:
Corrects benchmark test to iterate over the same value.

Signed-off-by: Vicente Zepeda Mas <chente.z.m@gmail.com>

chentex committed:
Fixing merge conflicts

Signed-off-by: Vicente Zepeda Mas <chente.z.m@gmail.com>

chentex committed:
Fixing duplicated test, left from merging changes

Signed-off-by: Vicente Zepeda Mas <chente.z.m@gmail.com>

chentex commented:
Changes done.
